src/Behat/Behat/ApplicationFactory.php

```diff
@@ -110,16 +110,9 @@ protected function getEnvironmentVariableName()
      */
     protected function getConfigPath()
     {
-        $cwd = rtrim(getcwd(), DIRECTORY_SEPARATOR);
-        $paths = array_filter(
-            array(
-                $cwd . DIRECTORY_SEPARATOR . 'behat.yml',
-                $cwd . DIRECTORY_SEPARATOR . 'behat.yml.dist',
-                $cwd . DIRECTORY_SEPARATOR . 'config' . DIRECTORY_SEPARATOR . 'behat.yml',
-                $cwd . DIRECTORY_SEPARATOR . 'config' . DIRECTORY_SEPARATOR . 'behat.yml.dist',
-            ),
-            'is_file'
-        );
+        $cwd = rtrim(getcwd(), DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR;
+        $configDir = 'config' . DIRECTORY_SEPARATOR;
+        $paths = glob("{$cwd}{,{$configDir}}behat.y{a,}ml{,.dist}", GLOB_BRACE);
 
         if (count($paths)) {
             return current($paths);
```

Comment:
this is now broken, as there is no guarantee that the glob will return the files in the order of precedence defined by Behat when there are several such files (for instance, when having both a committed ``behat.yml.dist`` and overwriting it with a git-ignored ``behat.yml``.

To limit the number of ``is_file`` calls, the solution could be to use a loop of the paths, do ``is_file`` inside the loop and then returning early as soon as a file is found (as we don't care about others anyway) instead of filtering the whole list